python-lib/dku_deeplearning_image/utils.py

```diff
@@ -77,25 +78,39 @@ def can_use_gpu():
     return len(tf.config.experimental.list_physical_devices('GPU')) > 0
 
 
-def set_gpu_options(should_use_gpu, gpu_list, memory_limit):
+def set_gpu_options(should_use_gpu, gpu_list, gpu_memory_allocation_mode, memory_limit_ratio=None):
     log_info("load_gpu_options")
     if should_use_gpu and can_use_gpu():
         log_info("should use GPU")
         gpus = tf.config.experimental.list_physical_devices('GPU')
+        for i, g in enumerate(gpus):
```

Comment:
Why is this needed?

Reply:
It is useful to map the coming from tf with function `tf.config.list_physical_devices('GPU')` to gpus from GPUtil. I then use these Ids in `calculate_gpu_memory_allocation(memory_limit_ratio, gpu_to_use)`

Reply:
oki